sdk/identity/identity/review/identity.api.md

```diff
@@ -185,6 +195,55 @@ export class ManagedIdentityCredential implements TokenCredential {
     getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null>;
     }
 
+// @public
+export abstract class MsalCredential implements TokenCredential {
+    constructor(msalFlow: MsalFlow, disableAutomaticAuthentication?: boolean | undefined);
+    // (undocumented)
+    protected disableAutomaticAuthentication?: boolean | undefined;
+    getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null>;
+    // (undocumented)
+    protected msalFlow: MsalFlow;
+}
+
+// @public
+export interface MsalCredentialOptions extends TokenCredentialOptions {
+    tokenCachePersistenceOptions?: TokenCachePersistenceOptions;
+}
+
+// @public
+export interface MsalFlow {
+    getActiveAccount(): Promise<AuthenticationRecord | undefined>;
+    getToken(scopes?: string[], options?: MsalGetTokenOptions): Promise<AccessToken | null>;
+    getTokenSilent(scopes?: string[], options?: MsalGetTokenOptions): Promise<AccessToken | null>;
+    init(options?: MsalGetTokenOptions): Promise<void>;
+    logout(): Promise<void>;
+}
+
+// @public
+export interface MsalGetTokenOptions extends GetTokenOptions {
+    // (undocumented)
+    correlationId?: string;
+    // (undocumented)
+    disableAutomaticAuthentication?: boolean;
+}
+
+// @public
+export abstract class MsalInteractiveCredential extends MsalCredential {
+    authenticate(scopes: string | string[], options?: GetTokenOptions): Promise<AuthenticationRecord | undefined>;
+}
+
+// @public
+export interface MsalInteractiveCredentialOptions extends MsalCredentialOptions {
+    authenticationRecord?: AuthenticationRecord;
+    disableAutomaticAuthentication?: boolean;
+}
```

Comment:
I'm not sure we want to add all these `Msal*` interfaces to the public API. Across all the Identity libraries we have avoided leaking MSAL APIs through our abstraction, keeping our dependence on it as an implementation detail. Because of this I would prefer we keep these classes out of the public API.

Reply:
I can hide them! I'll hide them tomorrow.

Reply:
Addressed here: https://github.com/Azure/azure-sdk-for-js/pull/14064/commits/cf0b35b7de9c5db9b38bd867767e4d923058dd03